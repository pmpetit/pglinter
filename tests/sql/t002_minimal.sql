-- Test for pglinter T002 rule: Tables with redundant indexes (minimal test)

--BEGIN;

DROP TABLE IF EXISTS test_redundant;

-- Create one simple test table
CREATE TABLE test_redundant (
    id INT PRIMARY KEY,
    first_name VARCHAR(100),
    last_name VARCHAR(100),
    tel TEXT
);

-- Create two identical indexes (redundant)
CREATE INDEX idx_name_1 ON test_redundant (first_name);
CREATE INDEX idx_name_2 ON test_redundant (first_name,last_name);
CREATE INDEX idx_name_3 ON test_redundant (first_name,last_name,tel);
CREATE INDEX idx_name_4 ON test_redundant (tel,last_name);

DROP EXTENSION IF EXISTS pglinter CASCADE;
CREATE EXTENSION IF NOT EXISTS pglinter;

-- Enable only T002
SELECT pglinter.disable_all_rules();
SELECT pglinter.enable_rule('T002');

-- Test T002
SELECT pglinter.perform_table_check();

-- Cleanup
--DROP TABLE test_redundant CASCADE;

--ROLLBACK;
