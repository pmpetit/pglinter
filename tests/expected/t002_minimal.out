-- Test for pglinter T002 rule: Tables with redundant indexes (minimal test)
\pset pager off
DROP TABLE IF EXISTS test_redundant;
NOTICE:  table "test_redundant" does not exist, skipping
-- Create one simple test table
CREATE TABLE test_redundant (
    id INT PRIMARY KEY,
    first_name VARCHAR(100),
    last_name VARCHAR(100),
    tel TEXT
);
-- Create two identical indexes (redundant)
CREATE INDEX idx_name_1 ON test_redundant (first_name);
CREATE INDEX idx_name_2 ON test_redundant (first_name,last_name);
CREATE INDEX idx_name_3 ON test_redundant (first_name,last_name,tel);
CREATE INDEX idx_name_4 ON test_redundant (tel,last_name);
DROP EXTENSION IF EXISTS pglinter CASCADE;
CREATE EXTENSION IF NOT EXISTS pglinter;
-- Enable only T002
SELECT pglinter.disable_all_rules();
NOTICE:  üî¥ Disabled 22 rule(s)
 disable_all_rules 
-------------------
                22
(1 row)

SELECT pglinter.enable_rule('T002');
NOTICE:  ‚úÖ Rule T002 has been enabled
 enable_rule 
-------------
 t
(1 row)

-- Test T002
SELECT pglinter.perform_table_check();
NOTICE:  üîç pglinter found 1 issue(s):
NOTICE:  ==================================================
NOTICE:  ‚ö†Ô∏è  [T002] WARNING: Found 3 redundant idx in table: 
public.test_redundant : idx_name_2 (first_name, last_name)) is a superset of idx_name_1 (first_name))
public.test_redundant : idx_name_3 (first_name, last_name, tel)) is a superset of idx_name_2 (first_name, last_name))
public.test_redundant : idx_name_3 (first_name, last_name, tel)) is a superset of idx_name_1 (first_name)) 

NOTICE:  ==================================================
NOTICE:  üìä Summary: 0 error(s), 1 warning(s), 0 info
NOTICE:  üü° Some warnings found - consider reviewing for optimization
 perform_table_check 
---------------------
 t
(1 row)

-- Cleanup
DROP TABLE test_redundant CASCADE;
--ROLLBACK;
